components/arm/xarm/xarm.go

```diff
@@ -152,6 +153,26 @@ func NewxArm(ctx context.Context, r robot.Robot, cfg config.Component, logger go
 	return &xA, nil
 }
 
+// UpdateAction helps hinting the reconfiguration process on what strategy to use given a modified config.
+// See config.UpdateActionType for more information.
+func (x *xArm) UpdateAction(c *config.Component) config.UpdateActionType {
+	remoteAddr := x.conn.RemoteAddr().String()
+	currentHost := string([]rune(remoteAddr)[:len(remoteAddr)-4])
```

Comment:
Could we add a comment specifying exactly what this is doing/why we are chopping the last four characters?

This is probably also an opportunity to put that `":502"` into a `const` called `defaultPort`, and then we can trim `len(defaultPort)` characters.